### ISSUE TITLE
Getting started tutorial documentation

### ISSUE THREAD

**Issue by tachyon1337** [reactjs/React.NET#447]
The documentation is out of date as it applies to Core 2.0. And for development environments other than windows, additional steps are likely necessary to get a default aspnet core template up and running using React.NET. 

Using Visual Studio for Mac on macOS sierra, I had to do the following to get the tutorial up and running.

(1) Install the following nuget package
`Install-Package JavaScriptEngineSwitcher.ChakraCore.Native.osx-x64`

(2) Fetch the latest Chakra runtime from github

[https://github.com/Microsoft/ChakraCore/releases](url)

and copy that over the old runtime installed by the nuget package

`.nuget/packages/javascriptengineswitcher.chakracore.native.osx-x64/2.4.6/runtimes/osx-x64/native/libChakraCore.dylib`

(3) add the following using statements to Startup.cs
```csharp
using JavaScriptEngineSwitcher.Core;
using JavaScriptEngineSwitcher.ChakraCore;
```

(4) Modify the return type of ConfigureServices
```csharp
public IServiceProvider ConfigureServices(IServiceCollection services)
{
      services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
      services.AddReact();
      services.AddMvc();
      return services.BuildServiceProvider();
}
```

(5) Switch out the javascript engine in Configure

```csharp
public void Configure(IApplicationBuilder app, IHostingEnvironment env)
{
      if (env.IsDevelopment())
      {
            app.UseDeveloperExceptionPage();
      }
      else
      {
             app.UseExceptionHandler("/Home/Error");
       }
      var engineSwitcher = JsEngineSwitcher.Instance;
      engineSwitcher.DefaultEngineName = ChakraCoreJsEngine.EngineName;
      engineSwitcher.EngineFactories
                       .AddChakraCore();
            
      app.UseReact(config=>{  
      });
      app.UseStaticFiles();
      app.UseMvc(routes =>
      {
            routes.MapRoute(
                 name: "default",
                  template: "{controller=Home}/{action=Index}/{id?}");
       });

}
```

Then the tutorial worked as expected. Posting the above might save someone else a number of hours scouring the issue threads.

**Comment by Taritsyn** [comment]
> (2) Fetch the latest Chakra runtime from github
>
> https://github.com/Microsoft/ChakraCore/releases
>
> and copy that over the old runtime installed by the nuget package
>
> .nuget/packages/javascriptengineswitcher.chakracore.native.osx-x64/2.4.6/runtimes/osx-x64/native/libChakraCore.dylib

This problem is solved in [version 3.0.0 Alpha 6](https://github.com/Taritsyn/JavaScriptEngineSwitcher/releases/tag/v3.0.0-alpha.6). There is even a [discussion on this topic](https://github.com/reactjs/React.NET/issues/355#issuecomment-336545711).

>      var engineSwitcher = JsEngineSwitcher.Instance;
>      engineSwitcher.DefaultEngineName = ChakraCoreJsEngine.EngineName;
>      engineSwitcher.EngineFactories
>                       .AddChakraCore();

If you install the [JavaScriptEngineSwitcher.Extensions.MsDependencyInjection](http://nuget.org/packages/JavaScriptEngineSwitcher.Extensions.MsDependencyInjection/) package, you can make this code better:

```csharp
services.AddJsEngineSwitcher(options =>
	options.DefaultEngineName = ChakraCoreJsEngine.EngineName
)
	.AddChakraCore()
	;
```

For more information, see in the [“Registration of JS engines > ASP.NET Core 1.X”](https://github.com/Taritsyn/JavaScriptEngineSwitcher/wiki/Registration-of-JS-engines#aspnet-core-1x) section of documentation.

**Comment by dustinsoftware** [comment]
Closing issues older than a year, please re-open if you think this is still relevant.
